File: service/src/main/kotlin/fi/espoo/evaka/shared/utils/Enums.kt
// SPDX-FileCopyrightText: 2017-2020 City of Espoo
//
// SPDX-License-Identifier: LGPL-2.1-or-later

package fi.espoo.evaka.shared.utils

inline fun <reified T : Enum<T>> parseEnum(value: String): T? {
    return try {
        enumValueOf<T>(value)
    } catch (e: IllegalArgumentException) {
        null
    }
}

File: frontend/src/lib-common/api-types/application/enums.ts
// SPDX-FileCopyrightText: 2017-2021 City of Espoo
//
// SPDX-License-Identifier: LGPL-2.1-or-later

export type ApplicationType = 'CLUB' | 'DAYCARE' | 'PRESCHOOL'

export type ApplicationStatus =
  | 'CREATED'
  | 'SENT'
  | 'WAITING_PLACEMENT'
  | 'WAITING_UNIT_CONFIRMATION'
  | 'WAITING_DECISION'
  | 'WAITING_MAILING'
  | 'WAITING_CONFIRMATION'
  | 'REJECTED'
  | 'ACTIVE'
  | 'CANCELLED'

export type ApplicationOrigin = 'ELECTRONIC' | 'PAPER'

export type ApplicationGuardianAgreementStatus =
  | 'AGREED'
  | 'NOT_AGREED'
  | 'RIGHT_TO_GET_NOTIFIED'

export type AttachmentType = 'URGENCY' | 'EXTENDED_CARE'

export type PlacementType = 
 | 'DAYCARE'
 | 'DAYCARE_PART_TIME'
